tiledb/api/src/array/attribute/mod.rs

```diff
@@ -728,6 +728,34 @@ mod test {
     use crate::filter::list::Builder as FilterListBuilder;
     use crate::filter::*;
 
+    /// Test what the default values filled in for `None` with attribute data are.
+    /// Mostly because if we write code which does need the default, we're expecting
+    /// to match core and need to be notified if something changes or we did something
+    /// wrong.
+    #[test]
+    fn attribute_defaults() {
+        let ctx = Context::new().expect("Error creating context instance.");
+
+        {
+            let spec = AttributeData {
+                name: "xkcd".to_owned(),
+                datatype: Datatype::UInt32,
+                ..Default::default()
+            };
+            let attr = spec.create(&ctx).unwrap();
+            assert_eq!(CellValNum::single(), attr.cell_val_num().unwrap());
+        }
+        {
+            let spec = AttributeData {
+                name: "xkcd".to_owned(),
+                datatype: Datatype::StringAscii,
+                ..Default::default()
+            };
+            let attr = spec.create(&ctx).unwrap();
+            assert_eq!(CellValNum::single(), attr.cell_val_num().unwrap());
```

Comment:
Huh. I trust CI on this, but I'm surprised by StringAscii defaulting to cell_val_num 1. Might be something worth filing a ticket for core on.

Reply:
I was also surprised, I wrote the test with `CellValNum::Var` expecting it to pass and it did not!

You do consider that a bug? I can see it from both sides